core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -0,0 +1,251 @@
+    @Macro
+    private Integer oracleServerPort;
+
+    @Description("Username to use to connect to the remote Oracle Host via SSH.")
+    @Macro
+    private String oracleServerUsername;
+
+    @Description("Password to use to connect to the remote Oracle Host via SSH.")
+    @Macro
+    private String oracleServerPassword;
+
+    @Description("Username to connect to the Oracle database.")
+    @Macro
+    private String dbUsername;
+
+    @Description("Password to connect the Oracle database.B")
+    @Macro
+    private String dbPassword;
+
+    @Description("Absolute path of the ``ORACLE_HOME`` environment variable on the Oracle server host." +
+      "This will be used to run the Oracle Spool utility.")
+    @Macro
+    private String oracleHome;
+
+    @Description("Oracle System ID(SID). This is used to uniquely identify a particular database on the system.")
+    @Macro
+    private String oracleSID;
+
+    @Description("Query to be executed for export." +
+      "For example: select * from test where name='cask';")
+    @Macro
+    private String queryToExecute;
+
+    @Description("Path to the directory where temporary SQL script needs to be created. It will be removed " +
+      "once the SQL command is executed.")
+    @Macro
+    private String tmpSQLScriptDirectory;
+
+    @Description("Path where output file will be exported.")
+    @Macro
+    private String outputPath;
+
+    @Description("Format of the output file. Acceptable values are csv, tsv, psv.")
+    @Macro
+    private String format;
+
+    public OracleExportActionConfig() {
+      this.oracleServerPort = 22;
+    }
+
+    public OracleExportActionConfig(String oracleServerHostname, @Nullable Integer oracleServerPort,
+                                    String oracleServerUsername, String oracleServerPassword,
+                                    String dbUsername, String dbPassword,
+                                    String oracleHome, String oracleSID,
+                                    String outputPath, String queryToExecute, String tmpSQLScriptDirectory,
+                                    String format) {
+      this.oracleServerHostname = oracleServerHostname;
+      this.oracleServerPort = oracleServerPort;
+      this.oracleServerUsername = oracleServerUsername;
+      this.oracleServerPassword = oracleServerPassword;
+      this.dbUsername = dbUsername;
+      this.dbPassword = dbPassword;
+      this.oracleHome = oracleHome;
+      this.oracleSID = oracleSID;
+      this.outputPath = outputPath;
+      this.queryToExecute = queryToExecute;
+      this.tmpSQLScriptDirectory = tmpSQLScriptDirectory;
+      this.format = format;
+    }
+
+    public void validate() {
+      if (!containsMacro("oracleServerPort") && oracleServerPort < 0) {
+        throw new IllegalArgumentException("Port cannot be negative");
+      }
+      if (!EnumUtils.isValidEnum(SeparatorFormat.class, format)) {
```

Comment:
Don't use ``EnumUtils`` to bring in unnecessary dependency. Can just do ``SeparatorFormat.valueOf(format)`` instead.